R/methods-Chromatograms.R

```diff
@@ -80,10 +89,18 @@ setMethod("[", "Chromatograms",
               ## Return a single element as a Chromatogram
               if (length(i) == 1 & length(j) == 1)
                   return(x@.Data[i, j, drop = TRUE][[1]])
+              pd <- x@phenoData
               ## Multiple elements, return type depends on drop.
               x <- x@.Data[i = i, j = j, drop = drop]
-              if (!drop)
+              if (!drop) {
                   x <- as(x, "Chromatograms")
+                  pd <- pd[j, ]
+                  ## Drop levels
+                  pData(pd) <- droplevels(pData(pd))
+                  ## set row names
+                  rownames(pd) <- colnames(x)
```

Comment:
It is required because:
```r
df <- data.frame(a = letters[1:4], idx = 1:4)
## rownames are not NULL
rownames(df)
[1] "1" "2" "3" "4"

## When I sub-set they keep the original row names.
df_sub <- df[2:3, ]
rownames(df_sub)
[1] "2" "3"

## That's no problem if we have *real* rownames set (the 1:nrow(df) above is the 
## case were we don't have rownames specified.
## The problem comes if we don't have rownames/colnames (e.g. a `Chromatogram` 
## without a `phenoData` specified). In that case, subsetting the matrix will 
## have e.g.
mat <- matrix(4, ncol = 4, nrow = 3)
mat[, 2:3]
     [,1] [,2]
[1,]    4    4
[2,]    4    4
[3,]    4    4

## And subsetting the corresponding `data.frame`/phenoData:
df[2:3, ]
  a idx
2 b   2
3 c   3
## With
rownames(df[2:3, ])
[1] "2" "3"

## See, I find that pretty misleading. In the case we don't have rownames/colnames 
## available the subsetted `data.frame` says we have rows with index 2 and 3, and
## the `matrix` says 1 and 2.
## Setting rownames(pd) <- colnames(x) does:
df_sub <- df[2:3, ]
rownames(df_sub) <- NULL
rownames(df_sub)
[1] "1" "2"

## So, for the case in which we do have colnames and rownames defined the assignment
## would not  be necessary, but when `is.null(colnames(x))` it is required.
```
Does that make sense?